cranelift/codegen/src/isa/x64/pcc.rs

```diff
@@ -0,0 +1,600 @@
+            dst,
+        } => {
+            let bits: u16 = size.to_bits().into();
+            let loaded = check_load(ctx, None, addr, vcode, size.to_type(), bits)?;
+            check_unop(ctx, vcode, 64, dst.to_writable_reg(), src1.into(), |src1| {
+                let sum = loaded.and_then(|loaded| ctx.add(src1, &loaded, bits));
+                clamp_range(ctx, 64, bits, sum)
+            })
+        }
+
+        Inst::AluRmiR {
+            size,
+            op: AluRmiROpcode::Sub,
+            src1,
+            src2: GprMemImm(RegMemImm::Imm { simm32 }),
+            dst,
+        } => {
+            let bits = size.to_bits().into();
+            check_unop(
+                ctx,
+                vcode,
+                64,
+                dst.to_writable_reg(),
+                src1.to_reg(),
+                |src1| {
+                    let simm32: i64 = simm32.into();
+                    clamp_range(ctx, 64, bits, ctx.offset(src1, bits, -simm32))
+                },
+            )
+        }
+
+        Inst::AluRmiR {
+            size,
+            src2: GprMemImm(RegMemImm::Mem { ref addr }),
+            dst,
+            ..
+        }
+        | Inst::AluRmRVex {
+            size,
+            src2: GprMem(RegMem::Mem { ref addr }),
+            dst,
+            ..
+        } => {
+            let loaded = check_load(ctx, None, addr, vcode, size.to_type(), 64)?;
+            check_output(ctx, vcode, dst.to_writable_reg(), &[], |_vcode| {
+                clamp_range(ctx, 64, size.to_bits().into(), loaded)
+            })
+        }
+        Inst::CmpRmiR {
+            size,
+            src: GprMemImm(RegMemImm::Mem { ref addr }),
+            ..
+        } => {
+            check_load(ctx, None, addr, vcode, size.to_type(), 64)?;
+            Ok(())
+        }
+
+        Inst::Cmove {
+            size,
+            consequent: GprMem(RegMem::Mem { ref addr, .. }),
+            ..
+        } => {
+            check_load(ctx, None, addr, vcode, size.to_type(), 64)?;
+            Ok(())
+        }
+
+        Inst::Imm { simm64, dst, .. } => {
+            check_output(ctx, vcode, dst.to_writable_reg(), &[], |_vcode| {
+                Ok(Fact::constant(64, simm64))
+            })
+        }
+
+        Inst::AluConstOp {
+            op: AluRmiROpcode::Xor,
+            dst,
+            ..
+        } => check_output(ctx, vcode, dst.to_writable_reg(), &[], |_vcode| {
+            Ok(Fact::constant(64, 0))
+        }),
+
+        Inst::AluRmiR {
+            size,
+            dst,
+            src2: GprMemImm(RegMemImm::Reg { .. } | RegMemImm::Imm { .. }),
+            ..
+        }
+        | Inst::AluConstOp { size, dst, .. }
+        | Inst::MovRR { size, dst, .. }
+        | Inst::AluRmRVex {
+            size,
+            src2: GprMem(RegMem::Reg { .. }),
+            dst,
+            ..
+        } => {
+            let bits: u16 = size.to_bits().into();
+            trace!("generic case: bits = {}", bits);
```

Comment:
Will all these traces in this file continue to be useful? As written, I don't think they really have enough context to be helpful, but maybe they would with more context in the message? Or maybe it would be better to just remove them?